frontend/src/views/dev_frame/load-mock-data.js

```diff
@@ -0,0 +1,227 @@
+                setNow(Math.round((count / total) * 100));
+            }
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    }
+
+    async function setupMockApplicants() {
+        try {
+            setStage("Applicants");
+            setNow(0);
+            count = 0;
+            total = mockApplicantData.length;
+            const data = normalizeImport(
+                {
+                    fileType: "json",
+                    data: mockApplicantData,
+                },
+                applicantSchema
+            );
+            for (let a of data) {
+                const applicant = await dispatch(upsertApplicant(a));
+                applicants.push(applicant);
+                count++;
+                setNow(Math.round((count / total) * 100));
+            }
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    }
+
+    async function setupMockAssignments() {
+        try {
+            setStage("Assignments");
+            setNow(0);
+            count = 0;
+            total = mockAssignmentData.length;
+            const data = normalizeImport(
+                {
+                    fileType: "json",
+                    data: mockAssignmentData,
+                },
+                assignmentSchema
+            ).map((p) =>
+                prepareFull.assignment(p, {
+                    positions,
+                    applicants,
+                    session,
+                })
+            );
+            for (let a of data) {
+                await dispatch(upsertAssignment(a));
+                count++;
+                setNow(Math.round((count / total) * 100));
+            }
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    }
+
+    async function clickHandler() {
+        try {
+            const sessions = await dispatch(fetchSessions());
+            if (sessions.some((s) => s.name === mockSessionData.name)) {
+                setDisable(true);
+                return;
+            }
+            setInProgress(true);
+            await setupMockSession();
+            await setupMockPositions();
+            await setupMockApplicants();
+            await setupMockAssignments();
+            setInProgress(false);
+        } catch (error) {
+            setInProgress(false);
+            return;
+        }
+    }
+
+    return (
+        <span
+            title={"One-click loads all mock data JSON files"}
+            className="mock-button"
+        >
+            <Button onClick={clickHandler}>Load Mock</Button>
+            <Modal show={disable} onHide={() => setDisable(false)} size="md">
+                <Modal.Header closeButton>
+                    <Modal.Title>{"Mock session already exists"}</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {"Please delete the mock session first before reloading."}
```

Comment:
What does this mean?

Reply:
If users have clicked the mock button, there will be a mock session already... so I set up a modal to remind that they have to delete the mock session (manually go to the session table and delete it) before they can click the button again...

Reply:
Let's randomly generate the session name based on a timestamp. That way there won't be conflicts.

Actually, it'd be really nice to have a "use existing session"/"new session" dropdown to decide where the data goes.

Reply:
Adding to existing sessions directly using the mock data may cause conflicts because it doesn't check diff/changed compared with existing data so that there could be duplicate position codes, assignments, and even applicants... But I think technically we could choose to compute diff first if using an existing session...

Reply:
No need. If it errors, it errors. This is a debug tools. We're going to assume the devs can understand the errors.